### PR TITLE
Follow the SBML structure of bounds and objectives

### PR DESCRIPTION
**Pull request by exaexa**
This moves the structs used in SBML.jl a bit closer to what is in actual SBML.
Functions that extract the bounds and objectives thus get slightly more
complicated. Hopefully this avoids interpretation mistakes.

Also, an attempt to revive Unitful ~~shall follow~~ follows.

### CODE REVIEW

**Review by paulflang — APPROVED** [review]
Looks good to me. SBMLtk ignores fbc and AFAIK also units. And I don't see anything else that would break it.

**Comment by exaexa** [comment]
> Looks good to me. SBMLtk ignores fbc and AFAIK also units. And I don't see anything else that would break it.

great thanks, I'll let this sink for a moment and merge ~tomorrow

**Comment by giordano** [comment]
On `master`:
```julia
julia> using SBML

julia> mdl = readSBML("e_coli_core.xml")
SBML.Model with 95 reactions, 72 species, and 5 parameters.

julia> flux_bounds(mdl)
([(0.0, "[fbc]"), (-1000.0, "[fbc]"), (-1000.0, "[fbc]"), (-1000.0, "[fbc]"), (0.0, "[fbc]"), (-1000.0, "[fbc]"), (0.0, "[fbc]"), (-1000.0, "[fbc]"), (-1000.0, "[fbc]"), (-1000.0, "[fbc]")  …  (-1000.0, "[fbc]"), (0.0, "[fbc]"), (-1000.0, "[fbc]"), (0.0, "[fbc]"), (0.0, "[fbc]"), (0.0, "[fbc]"), (-1000.0, "[fbc]"), (0.0, "[fbc]"), (-1000.0, "[fbc]"), (-1000.0, "[fbc]")], [(1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]")  …  (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]"), (1000.0, "[fbc]")])
```
with this PR:
```julia
julia> using SBML

julia> mdl = readSBML("e_coli_core.xml")
SBML.Model with 95 reactions, 72 species, and 5 parameters.

julia> flux_bounds(mdl)
([(-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, "")  …  (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, ""), (-Inf, "")], [(Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, "")  …  (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, ""), (Inf, "")])
```
I'm not sure yet what's happening, but it doesn't look right?

**Comment by exaexa** [comment]
hm yeah that's not good, looks like something is missing....

**Review by giordano — COMMENTED** [review]
Looks good overall, but we need to understand what's happening with the flux bounds

File: src/structs.jl [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+Compound units (such as "volt-amperes" and "dozens of yards per ounce") are
+built from multiple `UnitPart`s; see the definition of field `units` in
+[`SBML.Model`](@ref).
+"""
+struct UnitPart
+    kind::String
+    exponent::Int
+    scale::Int
+    multiplier::Float64
+    UnitPart(k, e, s, m) = new(k, e, s, m)
```
[the remark this answers is not in the document]
Comment (exaexa) [reply src/structs.jl]: true, thanks

**Comment by exaexa** [comment]
ok nvm let's merge. :rocket:

**Pull request merged**